entity-service/src/test/kotlin/com/egm/stellio/entity/web/EntityOperationHandlerTests.kt

```diff
@@ -521,4 +519,147 @@ class EntityOperationHandlerTests {
                 """.trimIndent()
             )
     }
+
+    @Test
+    fun `delete batch for all entities should return a 200 if JSON-LD payload is correct`() {
+        val entitiesIds = slot<List<String>>()
+        every { entityOperationService.splitEntitiesIdsByExistence(capture(entitiesIds)) } answers {
+            Pair(
+                entitiesIds.captured,
+                emptyList()
+            )
+        }
+        val canAdminEntitiesIds = slot<List<String>>()
+        every { authorizationService.splitEntitiesByUserCanAdmin(capture(canAdminEntitiesIds), any()) } answers {
+            Pair(
+                canAdminEntitiesIds.captured,
+                emptyList()
+            )
+        }
+
+        val computedEntitiesIdsToDelete = slot<Set<String>>()
+        every { entityOperationService.delete(capture(computedEntitiesIdsToDelete)) } returns
+            BatchOperationResult(
+                mutableListOf(
+                    "urn:ngsi-ld:Sensor:HCMR-AQUABOX1temperature",
+                    "urn:ngsi-ld:Sensor:HCMR-AQUABOX1dissolvedOxygen",
+                    "urn:ngsi-ld:Device:HCMR-AQUABOX1"
+                ),
+                mutableListOf()
+            )
+
+        val jsonLdFile = ClassPathResource("/ngsild/hcmr/HCMR_test_delete_all_entities.json")
+        webClient.post()
+            .uri("/ngsi-ld/v1/entityOperations/delete")
+            .header("Link", "<$aquacContext>; rel=http://www.w3.org/ns/json-ld#context; type=application/ld+json")
+            .bodyValue(jsonLdFile)
+            .exchange()
+            .expectStatus().isOk
+            .expectBody().json(
+                """
+                {
+                    "errors": [],
+                    "success": [
+                        "urn:ngsi-ld:Sensor:HCMR-AQUABOX1temperature",
+                        "urn:ngsi-ld:Sensor:HCMR-AQUABOX1dissolvedOxygen",
+                        "urn:ngsi-ld:Device:HCMR-AQUABOX1"
+                    ]
+                }
+                """.trimIndent()
+            )
+    }
+
+    @Test
+    fun `delete batch for unknown entities should return a 200 with explicit error messages`() {
+        val entitiesIds = slot<List<String>>()
+        every { entityOperationService.splitEntitiesIdsByExistence(capture(entitiesIds)) } answers {
+            Pair(
+                emptyList(),
+                entitiesIds.captured
+            )
+        }
+        val existingEntitiesIds = slot<List<String>>()
+        every { authorizationService.splitEntitiesByUserCanAdmin(capture(existingEntitiesIds), any()) } answers {
```

Comment:
you don't see to do something with the captured ids? (and indeed, the list should be empty, since they are all unknown)

Reply:
OK, I'll check them